src/gui/qgsmaptooldigitizefeature.h

```diff
@@ -17,31 +17,65 @@
 #define QGSMAPTOOLDIGITIZEFEATURE_H
 
 #include "qgsmaptoolcapture.h"
-#include "qgis_app.h"
+#include "qgis_gui.h"
 
 class QgsFeature;
 
-//! This tool digitizes geometry of new point/line/polygon features on already existing vector layers
-class APP_EXPORT QgsMapToolDigitizeFeature : public QgsMapToolCapture
+/**
+ * \ingroup gui
+ * \brief This tool digitizes geometry of new point/line/polygon features on already existing vector layers
+ * Once the map tool is enable, user can digitize the feature geometry
+ * A signal will then be emitted.
+ * \since QGIS 3.10
+ */
+class GUI_EXPORT QgsMapToolDigitizeFeature : public QgsMapToolCapture
 {
     Q_OBJECT
+
   public:
-    //! \since QGIS 3.2
-    QgsMapToolDigitizeFeature( QgsMapCanvas *canvas, QgsMapLayer *layer, CaptureMode mode );
+
+    /**
+     * \brief QgsMapToolDigitizeFeature is a map tool to digitize a feature geometry
+     * \param canvas the map canvas
+     * \param cadDockWidget widget to setup advanced digitizing parameters
+     * \param mode type of geometry to capture (point/line/polygon), QgsMapToolCapture::CaptureNone to autodetect geometry
+     */
+    QgsMapToolDigitizeFeature( QgsMapCanvas *canvas, QgsAdvancedDigitizingDockWidget *cadDockWidget, CaptureMode mode = QgsMapToolCapture::CaptureNone );
 
     void cadCanvasReleaseEvent( QgsMapMouseEvent *e ) override;
 
-    virtual void digitized( QgsFeature &f );
+    /**
+     * Change the layer edited by the map tool
+     * \param vl the layer to be edited by the map tool
+     * \since 3.10
+     */
+    void setLayer( QgsMapLayer *vl ) { mLayer = vl; }
```

Comment:
I'd move this implementation to the cpp file.

I'd expect some additional safety to be added in the future (e.g. cleaning up a rubberband if geometry type changes - or always, etc.). What it does is an implementation detail and not public API.